File: app/src/main/java/com/demo/mvp/findlocation/FindLocationContract.java
package com.demo.mvp.findlocation;

import android.location.Location;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.demo.mvp.BasePresenter;
import com.demo.mvp.BaseViewer;
import com.google.android.gms.common.api.Status;

interface FindLocationContract {
	interface Viewer extends BaseViewer {
		void showLocation();

		void setCurrentLocation(@Nullable Location location);

		void solveSettingDialogProblem(@NonNull Status status, int reqCode);

		void canNotShowSettingDialog();


	}

	interface Presenter extends BasePresenter {
		void findLocation();

		void afterLocationPermissionGranted();

		void afterLocationPermissionDenied();

		void release();

		Location getLastLocation();

		void stopFindingLocation();
	}
}

File: app/src/main/java/com/demo/mvp/findlocation/FindLocationContract.kt
package com.demo.mvp.findlocation

import android.app.Activity
import android.content.Context
import android.location.Location
import com.demo.mvp.BasePresenter
import com.demo.mvp.BaseViewer
import isochrone.isodistance.android.algorithm.TravelMode

interface FindLocationContract {
    interface Viewer : BaseViewer<Presenter> {
        fun getCurrentLocation()

        fun showCurrentLocation(location: Location?)

        fun canNotShowSettingDialog()

        fun getViewContext(): Context

        fun getViewActivity(): Activity

        fun showPolygon(travelMode: TravelMode, points: Array<isochrone.isodistance.android.domain.geocode.Location>)
    }

    interface Presenter : BasePresenter {
        fun findIsochrone(context: Context, target: isochrone.isodistance.android.domain.geocode.Location)

        fun findLocation()

        fun release()
    }
}